### PR TITLE
feat(providers): Update providers to return list of services for a service account

### PR DESCRIPTION
**Pull request by ksubrmnn**
Please describe the motivation for this PR and provide enough
information so that others can review it.

Providers should return all associated services for a service account, not just 1.

Please mark with X for applicable areas.

- New Functionality      [ ]
- Documentation          [ ]
- Install                [ ]
- Control Plane          [X]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests / CI System      [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
No

### CODE REVIEW

File: pkg/endpoint/providers/kube/client.go [hunk 1]
```diff
@@ -110,8 +110,8 @@ func (c Client) ListEndpointsForService(svc service.MeshService) []endpoint.Endp
 	return endpoints
 }
 
-// GetServiceForServiceAccount retrieves the service for the given service account
-func (c Client) GetServiceForServiceAccount(svcAccount service.K8sServiceAccount) (service.MeshService, error) {
+// GetServicesForServiceAccount retrieves the service for the given service account.
```
Comment (shashankram) [inline pkg/endpoint/providers/kube/client.go:113]:
```suggestion
// GetServicesForServiceAccount retrieves a list of services for the given service account
```

**Review by shashankram — DISMISSED** [review]

File: pkg/endpoint/providers/kube/client.go (hunk 1 above)
Comment (snehachhabria) [inline pkg/endpoint/providers/kube/client.go]:
```suggestion
// GetServicesForServiceAccount retrieves a list of services for the given service account.
```

Comment (ksubrmnn) [reply pkg/endpoint/providers/kube/client.go]: fixed

File: pkg/endpoint/providers/kube/client.go [hunk 2]
```diff
@@ -140,7 +140,8 @@ func (c Client) GetServiceForServiceAccount(svcAccount service.K8sServiceAccount
 				k8sServices, err := c.getServicesByLabels(selectorLabel, appNamspace)
 				if err != nil {
 					log.Error().Err(err).Msgf("Error retrieving service with label %v in namespace %s", selectorLabel, appNamspace)
-					return service.MeshService{}, errDidNotFindServiceForServiceAccount
+
+					return nil, errDidNotFindServiceForServiceAccount
```
Comment (snehachhabria) [inline pkg/endpoint/providers/kube/client.go:144]: Are we planning to return nil rather than empty slices going forward ?

Comment (ksubrmnn) [reply pkg/endpoint/providers/kube/client.go:144]: it does save an allocation. I don't feel strongly either way

**Pull request merged**